### PR TITLE
CURA-6915_remaining_issues

### PR DESCRIPTION
**Pull request by ninovanhooff**

### CODE REVIEW

File: plugins/UFPWriter/UFPWriter.py [hunk 1]
```diff
@@ -151,8 +156,30 @@ def _writeObjectList(archive):
...
+    @staticmethod
+    def _getObjectMetas(node: SceneNode) -> List[Dict]:
+        """Get object metadata to write for a Node.
+
+        :return: List of object metadata dictionaries.
+                 Might contain > 1 element in case of a group node.
+                 Might be empty in case of nonPrintingMesh
+        """
+
+        nodes = [node]
```
Comment (nallath) [inline plugins/UFPWriter/UFPWriter.py:178]: Why not use a DepthFirstIterator for this?

File: plugins/UFPWriter/UFPWriter.py [hunk 2]
```diff
@@ -151,8 +156,30 @@ def _writeObjectList(archive):
...
+        object_metas = []
+
+        for item in objects_model.items:
+            object_metas = object_metas + UFPWriter._getObjectMetas(item["node"])
 
         data = {METADATA_OBJECTS_PATH: object_metas}
         archive.setMetadata(data)
+
+    @staticmethod
+    def _getObjectMetas(node: SceneNode) -> List[Dict]:
```
Comment (nallath) [inline plugins/UFPWriter/UFPWriter.py:170]:
```suggestion
    def _getObjectMetas(node: SceneNode) -> List[Dict[str, str]]:
```

File: plugins/UFPWriter/UFPWriter.py [hunk 3]
```diff
@@ -151,8 +156,30 @@ def _writeObjectList(archive):
 
         To retrieve, use: `archive.getMetadata(METADATA_OBJECTS_PATH)`
         """
+
         objects_model = CuraApplication.getInstance().getObjectsModel()
-        object_metas = [{"name": item["name"]} for item in objects_model.items]
+        object_metas = []
+
+        for item in objects_model.items:
+            object_metas = object_metas + UFPWriter._getObjectMetas(item["node"])
```
Comment (nallath) [inline plugins/UFPWriter/UFPWriter.py:164]:
```suggestion
            object_metas.extend(UFPWriter._getObjectMetas(item["node"]))
```

File: plugins/UFPWriter/UFPWriter.py (hunk 1 above)
Comment (ninovanhooff) [reply plugins/UFPWriter/UFPWriter.py]: TIL

File: plugins/UFPWriter/UFPWriter.py [hunk 4]
```diff
@@ -151,8 +157,26 @@ def _writeObjectList(archive):
...
+        object_metas = []
+
+        for item in objects_model.items:
+            object_metas.extend(UFPWriter._getObjectMetas(item["node"]))
 
         data = {METADATA_OBJECTS_PATH: object_metas}
         archive.setMetadata(data)
+
+    @staticmethod
+    def _getObjectMetas(node: SceneNode) -> List[Dict[str, str]]:
```
Comment (fieldOfView) [inline plugins/UFPWriter/UFPWriter.py:171]: Might I suggest `_getObjectMetadata` instead? It is codestyle not to abbreviate, and a "meta" isn't a thing.

Comment (konskarm) [reply plugins/UFPWriter/UFPWriter.py]: Fixed in https://github.com/Ultimaker/Cura/pull/8023/commits/72310919c3748eb91832b1e5494161123a430e2d.

**Pull request merged**